packages/components/src/color-picker/inputs.js

```diff
@@ -15,54 +15,46 @@ import { DOWN, ENTER, UP } from '@wordpress/keycodes';
  * Internal dependencies
  */
 import IconButton from '../icon-button';
-import { isValidHex } from './utils';
 import TextControl from '../text-control';
 
 /* Wrapper for TextControl, only used to handle intermediate state while typing. */
-class Input extends Component {
-	constructor( { value } ) {
+export class Input extends Component {
+	constructor() {
 		super( ...arguments );
-		this.state = { value: String( value ).toLowerCase() };
 		this.handleBlur = this.handleBlur.bind( this );
 		this.handleChange = this.handleChange.bind( this );
 		this.handleKeyDown = this.handleKeyDown.bind( this );
 	}
 
-	componentWillReceiveProps( nextProps ) {
-		if ( nextProps.value !== this.props.value ) {
-			this.setState( {
-				value: String( nextProps.value ).toLowerCase(),
-			} );
-		}
-	}
-
 	handleBlur() {
-		const { valueKey, onChange } = this.props;
-		const { value } = this.state;
-		onChange( { [ valueKey ]: value } );
+		const { value, valueKey, onChange } = this.props;
+		onChange( {
+			state: 'commit',
+			[ valueKey ]: value,
+		} );
 	}
 
 	handleChange( value ) {
 		const { valueKey, onChange } = this.props;
-		// Protect against expanding a value while we're typing.
-		if ( value.length > 4 ) {
```

Comment:
Previous to this removal, the component will tell `ColorPicker` to update the other views (and the `onChangeComplete` callback) when the HEX value introduced was longer than 4 characters. I'm looking into port this behavior.

Reply:
Fixed in d06c6f9.